include/sendemail.php

```diff
@@ -463,31 +463,48 @@ function get_email_summary($buildid, $errors, $errorkey, $maxitems, $maxchars, $
         }
         $information .= "\n";
     } elseif ($errorkey == 'test_errors') {
-        $sql = '';
+        $tmpl = "
+            SELECT test.name, test.id
+            FROM build2test,test
+            WHERE build2test.buildid=?
+            AND test.id=build2test.testid
+            AND (build2test.status=?%s)
+            AND test.details NOT LIKE '%%Timeout%%'
+            ORDER BY test.id LIMIT $maxitems
+        ";
+
+        $or = '';
+        $params = [$buildid, 'failed'];
+
         if ($emailtesttimingchanged) {
-            $sql = 'OR timestatus>' . qnum($testtimemaxstatus);
+            $or = ' OR timestatus>?';
+            $params[] = $testtimemaxstatus;
         }
-        $test_query = pdo_query('SELECT test.name,test.id FROM build2test,test WHERE build2test.buildid=' . qnum($buildid) .
-            " AND test.id=build2test.testid AND (build2test.status='failed'" . $sql . ") ORDER BY test.id LIMIT $maxitems");
-        add_last_sql_error('sendmail');
-        $numrows = pdo_num_rows($test_query);
+
+        $sql = sprintf($tmpl, $or);
+        $pdo = get_link_identifier()->getPdo();
+        $query = $pdo->prepare($sql);
+
+        pdo_execute($query, $params);
+
+        $numrows = pdo_num_rows($query);
 
         if ($numrows > 0) {
             $information .= "\n\n*Tests failing*";
-            if (pdo_num_rows($test_query) == $maxitems) {
+            if (pdo_num_rows($query) == $maxitems) {
```

Comment:
I don't think pdo_num_rows() works for SELECT queries.  We could test it and find out for sure one way or the other.